### PR TITLE
Include recruitment_year into routes

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
`recruitment_cycle`

### Changes proposed in this pull request
Add `recruitment_cycle` into route

### Guidance to review
`http://localhost:3000/api/v1/2019/courses`
`http://localhost:3000/api/v1/2019/providers`
`http://localhost:3000/api/v1/2019/subjects`

### CODE REVIEW

**Comment by dankmitchell** [comment]
[Add recruitment_cycle parameter to the UCAS API](https://trello.com/c/HGWsgI55/873-add-recruitmentcycle-parameter-to-the-ucas-api)

File: app/models/course.rb [hunk 1]
```diff
@@ -57,4 +57,10 @@ class Course < ApplicationRecord
   has_and_belongs_to_many :subjects
   has_many :site_statuses
   has_many :sites, through: :site_statuses
+
+  def recruitment_year
+    "2019"
+  end
+
+  scope :recruitment_cycle, ->(year) { select { |c| c.recruitment_year == year } if year.present? }
```
Comment (misaka) [inline app/models/course.rb]: tests pls

File: app/serializers/course_serializer.rb [hunk 2]
```diff
@@ -57,8 +57,7 @@ def copy_form_required
     "Y" # we want to always create PDFs for applications coming in
   end
 
-  # TODO: make recruitment cycle dynamic
   def recruitment_cycle
-    "2019"
+    object.recruitment_year
```
Comment (misaka) [inline app/serializers/course_serializer.rb]: Is there a reason not to keep these two the same name?

File: app/models/course.rb [hunk 3]
```diff
@@ -57,4 +57,8 @@ class Course < ApplicationRecord
   has_and_belongs_to_many :subjects
   has_many :site_statuses
   has_many :sites, through: :site_statuses
+
+  def recruitment_cycle
+    "2019"
```
Comment (timabell) [inline app/models/course.rb]: should this be an int rather than a string given it's a year?

Comment (dankmitchell) [reply app/models/course.rb]: Docs say string and UCAS accepts a string

**Pull request merged**